test_echo.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+
+import pytest
+from echo_client import client_server
+import echo_server
+
+
+def test_HTTP_200():
+    response_first_line = client_server('''GET /index.html HTTP/1.1\r\n
+            Host: henryhowes.com\r\n
+            Content-Type: text/xml; charset=utf-8\r\n
+            ''').splitlines()[0].split()
+
+    assert response_first_line[1] == '200' and response_first_line[2] == 'OK'
+
+
+def test_HTTP_not_GET():
+    response_first_line = client_server('test').splitlines()[0].split()
+    assert response_first_line[1] == '405'
+
+
+def test_wrong_protocol():
+    response_first_line = client_server('''GET /index.html HTTP/1.0\r\n
+            Host: henryhowes.com\r\n
+            Content-Type: text/xml; charset=utf-8\r\n
+            ''').splitlines()[0].split()
+    assert response_first_line[1] == '505'
```

Comment:
These are all end-to-end functional tests of your server.  You should also have unit tests that verify that each function works the way it should.